packages/contracts-bedrock/src/L2/ETHLiquidity.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { WETH98 } from "src/dispute/weth/WETH98.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L1Block } from "src/L2/L1Block.sol";
+import { SafeSend } from "src/universal/SafeSend.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+import "src/libraries/errors/CommonErrors.sol";
+
+/// @title ETHLiquidity
+/// @notice The ETHLiquidity contract allows other contracts to access ETH liquidity without
+///         needing to modify the EVM to generate new ETH.
+contract ETHLiquidity {
+    /// @notice Emitted when an address burns ETH liquidity.
+    event LiquidityBurned(address indexed caller, uint256 value);
+
+    /// @notice Emitted when an address mints ETH liquidity.
+    event LiquidityMinted(address indexed caller, uint256 value);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0-beta.1";
+
+    /// @notice Allows an address to lock ETH liquidity into this contract.
+    function burn() external payable {
+        if (msg.sender != Predeploys.SUPERCHAIN_WETH) revert Unauthorized();
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
+        emit LiquidityBurned(msg.sender, msg.value);
+    }
+
+    /// @notice Allows an address to unlock ETH liquidity from this contract.
+    /// @param _amount The amount of liquidity to unlock.
+    function mint(uint256 _amount) external {
+        if (msg.sender != Predeploys.SUPERCHAIN_WETH) revert Unauthorized();
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
```

Comment:
We can inline the `IL1Block` interface in this contract if its easier

Reply:
Eh this is fine, compiler should handle it fine anyway. We can do this later if needed for 0.8.25 migration.